### ISSUE TITLE
Slither misses some obvious dependency relations

### ISSUE THREAD

**Issue by PriyankaBose** [crytic/slither#399]
```

pragma solidity ^0.4.24;

contract SlitherBug {
  uint256 rID_;
  bool public activated_ = false;

  modifier isHuman() {
    require(msg.sender == tx.origin, "sorry humans only");
    _;
  }
  
  // Function not vulnerable to re-entrancy attack

    function withdraw()
    isHuman()
    public {
        uint256 _rID = rID_;
  }
}

```

In the above contract slither outputs empty dependencies for the local varaible `_rID`. which is wrong. `_rID` is dependent on the state variable `_rID`.

My code to extract the dependency:

```
from slither.slither import Slither
import slither

slither_obj = Slither('slither_bug.sol')

for contract in slither_obj.contracts:
	if contract.name == 'SlitherBug':
		cont = contract

for function in cont.functions:
	if function.name == 'withdraw':
		func = function

node = func.nodes[1]

dep = slither.analyses.data_dependency.data_dependency.get_dependencies(node.irs[0].lvalue, func)

print(dep)
```

If I remove the line `require(msg.sender == tx.origin, "sorry humans only");` from the `modifier isHuman()` it outputs the state variable `_rID` as dependency; which is valid. 

Possibly a slither bug. Can you please look into this issue?

**Comment by montyly** [comment]
Hi @PriyankaBose, thank you for reporting the issue. 

I confirm that I can reproduce this behavior, I will look into it.

**Comment by montyly** [comment]
Hi @PriyankaBose, I fixed the bug in #400, it is accessible in `master` , and will be available in the upcoming new release. 

One issue is remaining though, Slither incorrectly classifies `withdraw()` as a protected function, as it has a check on `msg.sender`. We are looking at how to improve this heuristic behind this classficiation (https://github.com/crytic/slither/issues/401).

Let me know if you have any other issues, and feel free to join our [slack](https://empireslacking.herokuapp.com/) (#ethereum) if you need direct support.

**Comment by PriyankaBose** [comment]
Thanks a lot. Will let you know if I any further issues.
